### PR TITLE
pam: add optional faillock

### PR DESCRIPTION
**Pull request by valentindavid**
When `/etc/writable/faillock.enabled` is present, then we use `pam_faillock` which can lock accounts for 900 seconds after 3 wrong password.

Can be tested with `faillock --user username`. Also use `faillock --user username --reset` instead of waiting 900 seconds.

Requires https://github.com/snapcore/snapd/pull/12278

### CODE REVIEW

File: static/usr/lib/core/lockout-not-enabled.sh
Comment (mvo5) [inline static/usr/lib/core/lockout-not-enabled.sh:5]: This now needs to be `/etc/writable/account-locked.enabled` after the latest tweaks in https://github.com/snapcore/snapd/pull/12278

File: hook-tests/031-faillock.test
Comment (mvo5) [inline hook-tests/031-faillock.test:1]: (nitick) - this could be `/bin/sh` as well, right? Nothing bash specific in here?

File: hooks/031-faillock.chroot
Comment (mvo5) [inline hooks/031-faillock.chroot:1]: (same as above)

**Review by mvo5 — APPROVED** [review]
Thank you! One small tweak required but this looks good. Once successfully landed we should also port it to core20 :)

File: hook-tests/031-faillock.test
Comment (valentindavid) [reply hook-tests/031-faillock.test:1]: Why do we want `/bin/sh`?

Comment (mvo5) [reply hook-tests/031-faillock.test:1]: There is no deep reason, it's fine (hence the "nitpick"). It's just that on Ubuntu /bin/sh is dash which is slightly faster than bash but that also does not really matter. I guess it's a bit of a habit because Debian maintainer scripts are encouraged to run with /bin/sh (because there the speed actually adds up over thousands of sh runs from the maintainer scripts) and because in theory one could have a system without bash - but this is not really relevant here and as you can see all of my arguments are rather weak :) So fine to leave if you prefer that.

**Review by Meulengracht — APPROVED** [review]
LGTM!

File: hook-tests/031-faillock.test
Comment (valentindavid) [reply hook-tests/031-faillock.test:1]: This is a very good explanation. Thank you.

File: static/usr/lib/core/lockout-not-enabled.sh
Comment (valentindavid) [reply static/usr/lib/core/lockout-not-enabled.sh]: Good point. Done.

**Pull request merged**
